src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -531,6 +533,46 @@ public async Task RestoreDatabase()
                 if (ResourceNameValidator.IsValidResourceName(databaseName, ServerStore.Configuration.Core.DataDirectory.FullPath, out string errorMessage) == false)
                     throw new BadRequestException(errorMessage);
 
+                var extension = Path.GetExtension(restoreConfigurationJson.LastFileNameToRestore);
+                if (extension == Constants.Documents.PeriodicBackup.SnapshotExtension || extension == Constants.Documents.PeriodicBackup.EncryptedSnapshotExtension)
+                {
+                    long backupSizeInBytes;
+                    var zipPath = Path.Combine(restoreConfigurationJson.BackupLocation, restoreConfigurationJson.LastFileNameToRestore);
+
+                    try
+                    {
+                        using (var zip = ZipFile.OpenRead(zipPath))
+                            backupSizeInBytes = zip.Entries.Sum(entry => entry.Length);
+                    }
+                    catch (Exception e)
+                    {
+                        if (e is InvalidDataException)
+                            throw new InvalidDataException($"Invalid snapshot file: {restoreConfigurationJson.LastFileNameToRestore} at path : {zipPath}");
```

Comment:
I think we need to put `e` as inner exception in all of those exceptions here.

Reply:
Let's avoid multiple `:` in the message. Maybe change to sth like that:
```
$"Invalid snapshot file '{restoreConfigurationJson.LastFileNameToRestore}' at '{zipPath}' path"
```

This should isolate the path nicely (need to apply to other messages also sth similar)

Reply:
done